### PR TITLE
[Merged by Bors] - Documented service discovery

### PR DESCRIPTION
**Pull request by maltesander**
## Description

closes https://github.com/stackabletech/kafka-operator/issues/342

## Review Checklist
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/pages/discovery.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+= Discovery Profiles
+
+The Stackable Operator for Apache Kafka creates a single discovery profile that allows access to the Kafka cluster https://kubernetes.io/docs/concepts/services-networking/service/[`Service`] which is published into the Kubernetes cluster as a
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#configmap-v1-core[`ConfigMap`] object.
+
+== Profiles
+
+Given the following Kafka cluster:
+
+[subs="attributes"]
```
Comment (razvan) [inline docs/modules/ROOT/pages/discovery.adoc:14]:
```suggestion
[source,yaml,subs="attributes,callouts"]
```

File: docs/modules/ROOT/pages/discovery.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+metadata:
+  name: {clusterName} # <1>
+  namespace: {namespace} # <2>
+spec:
+  [...]
+----
+<1> Name of the Kafka cluster which is the name of the `ConfigMap` created for this discovery profile.
+<2> Namespace of the `ConfigMap` created for this discovery profile.
+
+The published `ConfigMap` name and namespace are equal to the cluster name and namespace. In this case `{namespace}/{clusterName}`.
```
Comment (razvan) [inline docs/modules/ROOT/pages/discovery.adoc:27]:
```suggestion
The published `ConfigMap` is created in the same namespace as the corresponding Apache Kafka cluster and has the same name. In this case `{namespace}/{clusterName}`.
```

File: docs/modules/ROOT/pages/discovery.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+=== Default
+
+This profile allows access to the Kafka cluster from inside the Kubernetes cluster, and connects directly to the metastore NodePort `Service`.
```
Comment (razvan) [inline docs/modules/ROOT/pages/discovery.adoc:31]:
```suggestion
This profile allows access to the Kafka cluster from inside the Kubernetes cluster, and connects directly to the metastore `NodePort` service.
```

File: docs/modules/ROOT/pages/discovery.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+== Contents
+
+The discovery profile contains the following fields where `{clusterName}` represents the name and `{namespace}` the namespace of the cluster:
+
+`KAFKA`::
+====
+Contains the thrift protocol connection string for the Kafka cluster to connect to the metastore:
+[subs="attributes"]
+  {clusterName}.{namespace}.svc.cluster.local:{brokerPort}
+====
```
Comment (razvan) [inline docs/modules/ROOT/pages/discovery.adoc:42]: I find this a bit more pleasing and understandable:
```suggestion
The discovery profile contains the thrift protocol connection string for the Kafka cluster.

In this example:
[subs="attributes"]
----
  {clusterName}.{namespace}.svc.cluster.local:{brokerPort}
----

```

**Review by razvan — COMMENTED** [review]
Some asciidoc fixes and suggestions.

File: docs/modules/ROOT/pages/discovery.adoc (hunk 1 above)
Comment (maltesander) [reply docs/modules/ROOT/pages/discovery.adoc]: done

**Review by razvan — APPROVED** [review]
lgtm

**Comment by maltesander** [comment]
bors merge

**Pull request closed without merging**